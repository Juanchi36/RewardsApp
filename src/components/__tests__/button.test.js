import React from 'react';
import { render } from '@testing-library/react-native';
import { Button } from '@components';

describe('button', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return a snapshot of the component', () => {
    const result = render(<Button title="A.TITLE" />);

    const button = result.getByTestId('primary-button');
    console.log(button.props.style);

    expect(button).toBeDefined();
    expect(result.toJSON()).toMatchSnapshot();
  });

  it('should return a blue button', () => {
    const result = render(<Button title="A.TITLE" />);

    const button = result.getByTestId('primary-button');
    console.log(button.props.style);

    expect(button.props.style[0].backgroundColor).toBe('#334ffa');
    expect(result.toJSON()).toMatchSnapshot();
  });
});
