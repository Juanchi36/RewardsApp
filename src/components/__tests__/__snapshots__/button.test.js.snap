// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should return a blue button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#334ffa",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
        },
      ]
    }
    testID="primary-button"
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "800",
            "lineHeight": 21.86,
          },
        ]
      }
    >
      A.TITLE
    </Text>
  </View>
</View>
`;

exports[`button should return a snapshot of the component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#334ffa",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
        },
      ]
    }
    testID="primary-button"
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "800",
            "lineHeight": 21.86,
          },
        ]
      }
    >
      A.TITLE
    </Text>
  </View>
</View>
`;
